kbfssync/semaphore_test.go

```diff
@@ -131,19 +131,19 @@ func TestCancel(t *testing.T) {
 	s := NewSemaphore()
 	require.Equal(t, int64(0), s.countForTest())
 
+	// Do this before spawning the goroutine to avoid a race on
+	// the count field of the result of callAcquire.
```

Comment:
You mean `acquireCall`, not `callAcquire`, right?

But still I don't really understand how this comment relates to the race.  There seems to be only one writer of the field (when the `acquireCall` is constructed in `callAcquire`), and then it gets put on a channel, and the `count` field isn't read until it comes out of the channel.

The race is actually that the `count` returned by `Semaphore.Acquire` is racy, depending on whether the `select` acts on the `Release()` or the context cancellation first.  Right?  By moving this before the `Acquire`, you guarantee that the semaphore's internal count will be increased before the `tryAcquire` call happens.